alacritty_terminal/src/term/mod.rs

```diff
@@ -2429,6 +2736,288 @@ mod tests {
+        term.move_forward(Column(term.columns()));
+        term.damage.reset(num_cols);
+
+        term.move_backward_tabs(1);
+        assert_eq!(term.damage.lines[7], LineDamageBounds { line: 7, left: 8, right });
+        term.save_cursor_position();
+        term.goto(Line(1), Column(1));
+        term.damage.reset(num_cols);
+
+        term.restore_cursor_position();
+        assert_eq!(term.damage.lines[1], LineDamageBounds { line: 1, left: 1, right: 1 });
+        assert_eq!(term.damage.lines[7], LineDamageBounds { line: 7, left: 8, right: 8 });
+        term.damage.reset(num_cols);
+
+        term.clear_line(ansi::LineClearMode::All);
+        assert_eq!(term.damage.lines[7], LineDamageBounds { line: 7, left: 0, right });
+        term.damage.reset(num_cols);
+
+        term.clear_line(ansi::LineClearMode::Left);
+        assert_eq!(term.damage.lines[7], LineDamageBounds { line: 7, left: 0, right: 8 });
+        term.damage.reset(num_cols);
+
+        term.clear_line(ansi::LineClearMode::Right);
+        assert_eq!(term.damage.lines[7], LineDamageBounds { line: 7, left: 8, right });
+        term.damage.reset(num_cols);
+
+        term.reverse_index();
+        assert_eq!(term.damage.lines[7], LineDamageBounds { line: 7, left: 8, right: 8 });
+        assert_eq!(term.damage.lines[6], LineDamageBounds { line: 6, left: 8, right: 8 });
+    }
+
+    #[test]
+    fn damage_vi_movements() {
+        let size = SizeInfo::new(10.0, 10.0, 1.0, 1.0, 0.0, 0.0, false);
+        let mut term = Term::new(&Config::default(), size, ());
+        let num_cols = term.columns();
+        // New termianal is always damaged. Reset it since we test when damage is partial.
+        term.reset_damage();
+
+        term.toggle_vi_mode();
+        assert!(term.damage.is_fully_damaged);
+        term.reset_damage();
+
+        // NOTE Since [`Term::damage`] always includes cursor we'll be comparing damaged lines
+        // directly, since we're testing that all functions bellow damage cursor in right places.
+        term.vi_goto_point(Point::new(Line(5), Column(5)));
+        assert_eq!(term.damage.lines[0], LineDamageBounds { line: 0, left: 0, right: 0 });
+        assert_eq!(term.damage.lines[5], LineDamageBounds { line: 5, left: 5, right: 5 });
+        term.damage.reset(num_cols);
+
+        term.vi_motion(ViMotion::Up);
+        term.vi_motion(ViMotion::Right);
+        term.vi_motion(ViMotion::Up);
+        term.vi_motion(ViMotion::Left);
+        assert_eq!(term.damage.lines[3], LineDamageBounds { line: 3, left: 5, right: 6 });
+        assert_eq!(term.damage.lines[4], LineDamageBounds { line: 4, left: 5, right: 6 });
+        assert_eq!(term.damage.lines[5], LineDamageBounds { line: 5, left: 5, right: 5 });
+
+        // Ensure that we haven't damaged entire terminal during the test.
+        assert!(!term.damage.is_fully_damaged);
+    }
+
+    #[test]
+    fn should_be_fully_damaged() {
```

Comment:
```suggestion
    fn full_damage() {
```

Some tests in here test that things are **not** fully damaged. I think `full_damage` thus fits a little better as a name.